Allow embedding parameters into firmware

Williangalvani opened the pull request:
supports injecting them into the .apj firmware for stm32 boards, and creating a .params file in userdata for linux

https://github.com/bluerobotics/BlueOS-docker/assets/4013804/2a471371-c7af-4c92-9682-41fe34638f10

ps: I just increased the width of the select so the full label shows up

additional fix for #1734

this can be tested by running the wizard and chosing a default set, and then resetting the parameters to firmware defaults in qgc and checking if the configuration remains (easier to see frame type in the 3d viewer, for sub at least)

fix #1720 
fix #1722 
fix #1726

joaoantoniocardoso reviewed (changes requested):
Not gracefully, but it works. 

To fix - after 'retry' is clicked, the list keeps showing the last error:

https://github.com/bluerobotics/BlueOS-docker/assets/5920286/9c8980ba-5ec2-475f-b4ae-c98524655a18

To fix - available param options are weird, showing the wrong list for the chosen vehicle type..?

https://github.com/bluerobotics/BlueOS-docker/assets/5920286/f5153565-8047-49f0-a86e-39e4962d951f

To fix - it shows "configuration was aborted" as if it was an error, but it was the user's choice:

https://github.com/bluerobotics/BlueOS-docker/assets/5920286/d92a0c84-af00-4405-a9fb-68738007dec7

To fix (maybe) - annoyingly weird scroll here (Firefox), stops before the end:

https://github.com/bluerobotics/BlueOS-docker/assets/5920286/820715fa-c13e-4ece-ae86-d655cda65573

To check - cancel while uploading may cause harm (Pixhawk):

https://github.com/bluerobotics/BlueOS-docker/assets/5920286/4703a0f8-ee91-4648-b84b-07873963e2fc

Williangalvani commented:
> To fix - after 'retry' is clicked, the list keeps showing the last error:

fixed

> To fix - available param options are weird, showing the wrong list for the chosen vehicle type..?

added cleanup when selecting a vehicle, for people who like clicking "back"

> To check - cancel while uploading may cause harm (Pixhawk):

it can't really, as the process in the backend cannot be stopped

> To fix - it shows "configuration was aborted" as if it was an error, but it was the user's choice:

![image](https://github.com/bluerobotics/BlueOS-docker/assets/4013804/462ae878-68e7-461f-b09a-21b6b1f68075)

joaoantoniocardoso reviewed (commented):
* To fix: disable cancel during the upload, otherwise it behaves unstable:

https://github.com/bluerobotics/BlueOS-docker/assets/5920286/97ca58f1-037b-4c31-9b21-3c87a99f2c5a


* To fix: remove the cancel button after the successfully uploaded:

https://github.com/bluerobotics/BlueOS-docker/assets/5920286/0e9ef26c-1b85-4177-95f5-d3b669a7983e

Williangalvani commented:
> To fix: disable cancel during the upload, otherwise it behaves unstable:

interesting, I think the ardupilotmanager is "stuck" flashing and doesn't handle other requests.

well. I just removed the cancel button. we can't reliably cancel the promises anyway.

joaoantoniocardoso reviewed (commented):
Just one thing missing: this combo box should be disabled until the data is loaded because the user can't know if it's not loaded yet or really there are no parameters. Best if it's possible to show a loading message:

https://github.com/bluerobotics/BlueOS-docker/assets/5920286/e08f9754-a84d-4eb0-9d31-d4508af50ac0

joaoantoniocardoso reviewed (approved):
:100:
